### PR TITLE
Ca reload

### PR DESCRIPTION
**Pull request by george-angel**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -23,7 +23,7 @@ Usage of /kube-service-mirror:
   -svc-sync
         sync services on startup (default true)
   -target-kube-config string
-        (required) Path of the target cluster kube config file to mirrot services from
+        Path of the target cluster kube config file to mirrot services from
```
Comment (ribbybibby) [inline README.md:26]:
```suggestion
        Path of the target cluster kube config file to mirror services from
```

File: main.go [hunk 2]
```diff
@@ -4,21 +4,30 @@ import (
...
 	flagLogLevel             = flag.String("log-level", "info", "Log level, defaults to info")
 	flagResyncPeriod         = flag.Duration("resync-period", 60*time.Minute, "Namespace watcher cache resync period")
 	flagMirrorNamespace      = flag.String("mirror-ns", "", "The namespace to create dummy mirror services in")
 	flagSvcPrefix            = flag.String("svc-prefix", "", "(required) A prefix to apply on all mirrored services names. Will also be used for initial service sync")
 	flagLabelSelector        = flag.String("label-selector", "", "(required) Label of services and endpoints to watch and mirror")
 	flagSvcSync              = flag.Bool("svc-sync", true, "sync services on startup")
+
+	saToken = os.Getenv("SERVICE_ACCOUNT_TOKEN")
+	apiURL  = os.Getenv("KUBE_API_SERVER")
+	caURL   = os.Getenv("REMOTE_CLUSTER_CA_CERT_URL")
```
Comment (ribbybibby) [inline main.go:28]:
```suggestion
	saToken = os.Getenv("REMOTE_SERVICE_ACCOUNT_TOKEN")
	apiURL  = os.Getenv("REMOTE_KUBE_API_SERVER")
	caURL   = os.Getenv("REMOTE_CLUSTER_CA_CERT_URL")
```
Prefix all of these values with `REMOTE` to help communicate their usage?

Also, I don't like the inconsistent mixture of env vars and flags. I'd prefer flags for everything, env vars or both for all options. The token could be a path to a file so it isn't exposed in the flags?

File: main.go [hunk 3]
```diff
@@ -14,18 +17,19 @@ import (
...
-	flagMirrorNamespace      = flag.String("mirror-ns", "", "The namespace to create dummy mirror services in")
-	flagSvcPrefix            = flag.String("svc-prefix", "", "(required) A prefix to apply on all mirrored services names. Will also be used for initial service sync")
-	flagLabelSelector        = flag.String("label-selector", "", "(required) Label of services and endpoints to watch and mirror")
-	flagSvcSync              = flag.Bool("svc-sync", true, "sync services on startup")
+	flagMirrorNamespace      = flag.String("mirror-ns", getEnv("KSM_MIRROR_NS", ""), "The namespace to create dummy mirror services in")
+	flagSvcPrefix            = flag.String("svc-prefix", getEnv("KSM_SVC_PREFIX", ""), "(required) A prefix to apply on all mirrored services names. Will also be used for initial service sync")
+	flagLabelSelector        = flag.String("label-selector", getEnv("KSM_LABEL_SELECTOR", ""), "(required) Label of services and endpoints to watch and mirror")
+	flagSvcSync              = flag.Bool("svc-sync", true, "Sync services on startup")
+	flatRemoteAPIURL         = flag.String("remote-api-url", getEnv("KSM_REMOTE_API_URL", ""), "Remote Kubernetes API server URL")
```
Comment (ribbybibby) [inline main.go:28]: `flagRemoteAPIURL`*

File: main.go [hunk 4]
```diff
@@ -75,7 +112,7 @@ func main() {
 	if *flagTargetKubeConfigPath != "" {
 		remoteClient, err = kube.ClientFromConfig(*flagTargetKubeConfigPath)
 	} else {
-		remoteClient, err = kube.Client(saToken, apiURL, caURL)
+		remoteClient, err = kube.Client(saToken, *flatRemoteAPIURL, *flagRemoteCAURL)
```
Comment (ribbybibby) [inline main.go:115]: `*flagRemoteAPIURL`*

**Review by ribbybibby — APPROVED** [review]

**Review by ffilippopoulos — APPROVED** [review]

**Pull request merged**